web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.Header.tsx

```diff
@@ -107,31 +87,40 @@ const RegistryLayoutHeader: React.FC = () => {
         pathname={pathname}
         extras={
           <Box display="flex" justifyContent="center" alignItems="center">
-            {chainId && loaded && isConnected && disconnect && (
-              <UserMenuItems
-                address={truncate(wallet?.address)}
-                avatar={party?.image ? party?.image : defaultAvatar}
-                disconnect={disconnect}
-                pathname={pathname}
-                linkComponent={RegistryNavLink}
-                userMenuItems={userMenuItems}
-                profiles={
-                  partiesByAccountId?.accountById?.partiesByAccountId?.nodes?.map(
-                    party => ({
-                      name: party?.name ? party?.name : DEFAULT_NAME,
-                      profileImage: party?.image
-                        ? party?.image
-                        : getDefaultAvatar(party),
-                      address: truncate(party?.walletByWalletId?.addr),
+            {chainId &&
+              ((!authLoading && !!activeAccount) ||
+                (walletLoaded && isConnected)) &&
+              disconnect && (
+                <UserMenuItems
+                  address={getAddress({
+                    walletAddress: wallet?.address,
+                    email: activeAccount?.email,
+                  })}
+                  avatar={
+                    activeAccount?.image ? activeAccount?.image : defaultAvatar
+                  }
+                  disconnect={disconnect}
+                  pathname={pathname}
+                  linkComponent={RegistryNavLink}
+                  userMenuItems={userMenuItems}
+                  profiles={
+                    authenticatedAccounts?.map(account => ({
+                      name: account?.name ? account?.name : DEFAULT_NAME,
+                      profileImage: account?.image
+                        ? account?.image
+                        : getDefaultAvatar(account),
+                      address: getAddress({
+                        walletAddress: account?.addr,
+                        email: account?.email,
+                      }),
                       selected:
-                        wallet?.address === party?.walletByWalletId?.addr,
-                    }),
-                  ) || []
-                }
-                addAddress={handleAddAddress}
-                onProfileClick={onProfileClick}
-              />
-            )}
+                        activeAccount?.id && activeAccount?.id === account?.id,
```

Comment:
Not sure if it's possible but if we have `0` as an id the condition would always fail.

Reply:
I don't think this is possible ids when defined are always uuids